api/envoy/api/v2/endpoint/load_report.proto

```diff
@@ -116,6 +160,9 @@ message ClusterStats {
   repeated UpstreamLocalityStats upstream_locality_stats = 2
       [(validate.rules).repeated .min_items = 1];
 
+  // Endpoint granularity stats per downstream host based on incoming requests.
+  repeated DownstreamEndpointStats downstream_endpoint_stats = 6;
+
```

Comment:
Lets not pollute the ClusterStats with this information, we probably need a separate message 

message ServerClusterStats {
  // Endpoint granularity stats per downstream host based on incoming requests.
  repeated DownstreamEndpointStats downstream_endpoint_stats = 6;
}

and then add a new service 

ServerLoadStatsRequest in lrs.proto which uses this message for sending load report to management server.

Reply:
+1. Not sure about `ServerLoadStats` or `ServerClusterStats` though. It's really downstream oriented I think and not by cluster.